File: backend/app/schemas/clothing.py
from uuid import UUID
from datetime import datetime
from pydantic import BaseModel


class ClothingItemResponse(BaseModel):
    id: UUID
    user_id: UUID
    image_url: str | None = None
    name: str | None
    category: str
    subcategory: str | None
    primary_color: str | None
    secondary_colors: list[str] | None
    style_tags: list[str] | None
    occasion_tags: list[str] | None = None
    season_tags: list[str] | None
    pattern: str | None = None
    silhouette: str | None = None
    warmth_level: int | None
    temp_range_min: int | None = None
    temp_range_max: int | None = None
    brand: str | None
    notes: str | None
    is_active: bool
    classification_raw: dict | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class ClothingItemUpdate(BaseModel):
    name: str | None = None
    brand: str | None = None
    notes: str | None = None
    category: str | None = None
    subcategory: str | None = None
    style_tags: list[str] | None = None
    occasion_tags: list[str] | None = None
    season_tags: list[str] | None = None
    pattern: str | None = None
    silhouette: str | None = None
    warmth_level: int | None = None
    temp_range_min: int | None = None
    temp_range_max: int | None = None

File: frontend/src/api/clothing.ts
import api from './client'

export interface ClothingItem {
  id: string
  user_id: string
  image_url: string | null
  category: string
  subcategory: string | null
  primary_color: string | null
  secondary_colors: string[] | null
  style_tags: string[] | null
  season_tags: string[] | null
  warmth_level: number | null
  brand: string | null
  notes: string | null
  is_active: boolean
  created_at: string
}

export const clothingApi = {
  upload: (file: File) => {
    const form = new FormData()
    form.append('file', file)
    return api.post<ClothingItem>('/clothing/upload', form, {
      headers: { 'Content-Type': 'multipart/form-data' },
    })
  },

  list: (params?: { category?: string; season?: string; is_active?: boolean; page?: number; limit?: number }) =>
    api.get<ClothingItem[]>('/clothing/', { params }),

  get: (id: string) => api.get<ClothingItem>(`/clothing/${id}`),

  update: (id: string, data: Partial<Pick<ClothingItem, 'brand' | 'notes' | 'style_tags' | 'season_tags' | 'category' | 'subcategory'>>) =>
    api.patch<ClothingItem>(`/clothing/${id}`, data),

  delete: (id: string) => api.delete(`/clothing/${id}`),

  toggleArchive: (id: string) => api.post<ClothingItem>(`/clothing/${id}/archive`),
}
